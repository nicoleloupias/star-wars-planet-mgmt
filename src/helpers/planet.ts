import { Planet } from "../services/types/Planet";

export const getPlanetId = (url: Planet["url"]) => {
  const regex = /\/planets\/(\d+)\//;
  const test = url.match(regex);
  if (!test) return -1;
  return test[1];
};

const DEFAULT_PLANET_COLOR = "gray.300";

export const getRandomPlanetColor = (name?: string) => {
  if (!name) return DEFAULT_PLANET_COLOR;
  const stringUniqueHash = [...name].reduce((acc, char) => {
    return char.charCodeAt(0) + ((acc << 5) - acc);
  }, 0);
  return `hsl(${stringUniqueHash % 360}, 35%,  45%)`;
};
