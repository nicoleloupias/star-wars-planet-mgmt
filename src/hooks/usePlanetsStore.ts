import { create } from "zustand";
import { Planet } from "../services/types/Planet";

interface PlanetStore {
  planets: Planet[];
  setPlanets: (planets: Planet[]) => void;
}

export const usePlanetsStore = create<PlanetStore>((set, get) => ({
  planets: [],
  setPlanets: (planets) => {
    return set((state) => {
      return { ...state, planets: [...state.planets, ...planets] };
    });
  }
}));
