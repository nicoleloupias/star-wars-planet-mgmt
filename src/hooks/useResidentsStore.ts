import { create } from "zustand";
import type { Planet } from "./usePlanetsStore";
import type { Resident } from "../services/types/Resident";

interface ResidentsStore {
  residents: Record<Planet["id"], Resident[]>;
  setResident: (planetId: Planet["id"], resident: Resident) => void;
}

export const useResidentsStore = create<ResidentsStore>((set) => ({
  residents: {},
  setResident: (planetId, resident) => {
    return set((state) => {
      return {
        ...state,
        residents: {
          ...state.residents,
          [planetId]: [...(state.residents?.[planetId] || []), resident]
        }
      };
    });
  }
}));
