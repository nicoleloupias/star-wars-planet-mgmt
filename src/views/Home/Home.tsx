import { Box, Center, Flex, SimpleGrid, Text } from "@chakra-ui/react";
import { PlanetCard } from "./components/PlanetCard";
import type { Planet } from "../../hooks/usePlanetsStore";
import { usePlanetsStore } from "../../hooks/usePlanetsStore";
import { Hero } from "./components/Hero";
import { useEffect, useState } from "react";
import { useForm } from "react-hook-form";
import { SortBy } from "./components/SortBy";
import { SearchBar } from "./components/SearchBar";

export interface FormFieldValues {
  search: string;
  sortBy: "name" | "diameter" | "climate" | "terrain" | "population";
}

export const Home = () => {
  const planets = usePlanetsStore((state) => state.planets);
  const [planetsToShow, setPlanetsToShow] = useState<Planet[] | undefined>();
  const form = useForm<FormFieldValues>();

  useEffect(() => {
    if (planets) {
      setPlanetsToShow([...planets]);
    }
  }, [planets]);

  return (
    <>
      <Hero />
      <Center flexDir="column">
        <Box as="section" minW={{ lg: "1040px" }} mt={10}>
          <Flex as="form" gap={2} justifyItems="flex-start" flexDir={{ base: "column", lg: "row" }}>
            <SearchBar setPlanetsToShow={setPlanetsToShow} form={form} />
            <SortBy setPlanetsToShow={setPlanetsToShow} planetsToShow={planetsToShow} form={form} />
          </Flex>

          <SimpleGrid my={20} columns={{ base: 1, md: 2, lg: 3 }} gap={10} justifyItems="center">
            {planetsToShow && planetsToShow?.map((planet) => <PlanetCard key={planet.id} {...planet} />)}
            {form.watch("search") && planetsToShow?.length === 0 && <Text>There are no results with this search</Text>}
          </SimpleGrid>
        </Box>
      </Center>
    </>
  );
};
