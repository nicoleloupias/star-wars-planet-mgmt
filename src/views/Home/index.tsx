import { useEffect, useState } from "react";
import { PlanetService } from "../../services/PlanetService";
import { Box, Button, Center, Grid, GridItem, SimpleGrid } from "@chakra-ui/react";
import { PlanetCard } from "../../components/PlanetCard";
import { usePlanetsStore } from "../../hooks/usePlanetsStore";

export const Home = () => {
  const planets = usePlanetsStore((state) => state.planets);
  const setPlanets = usePlanetsStore((state) => state.setPlanets);
  const getAllPlanets = async () => {
    const planets = await PlanetService.getAll();

    setPlanets(planets);
  };

  useEffect(() => {
    if (planets.length === 0) {
      getAllPlanets();
    }
  }, []);

  return (
    <Center>
      <SimpleGrid mt={20} columns={{ base: 1, md: 2, lg: 3 }} gap={10}>
        {planets.length > 0 && planets?.map((planet) => <PlanetCard key={planet.url} {...planet} />)}
      </SimpleGrid>
    </Center>
  );
};
