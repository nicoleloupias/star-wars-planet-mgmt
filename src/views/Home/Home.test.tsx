import { beforeAll, describe, expect, test } from "vitest";
import { Home } from "./Home";
import { render } from "../../test-utils";
import { usePlanetsStore } from "../../hooks/usePlanetsStore";
import { act, fireEvent, waitFor } from "@testing-library/react";
import { planetsMock } from "../../mocks/planet";

describe("<Home/>", () => {
  beforeAll(() => {
    usePlanetsStore.setState(
      {
        planets: planetsMock
      },

      true
    );
  });
  test("should render succesfully", () => {
    const component = render(<Home />);

    expect(component).toMatchSnapshot();
  });

  test("should render a card for each planet", () => {
    const component = render(<Home />);

    expect(component.getAllByTestId("planet-card").length).toBe(10);
  });

  test("should render 'there are no results' if planetsToShow is 0", async () => {
    const component = render(<Home />);

    act(() => {
      fireEvent.change(component.getByTestId("search"), { target: { value: "asdf" } });
      fireEvent.click(component.getByRole("button", { name: "Search" }));
    });

    await waitFor(() => {
      expect(component.getByText("There are no results with this search")).toBeInTheDocument();
    });
  });

  test("should paginate correctly on next button click", async () => {
    const component = render(<Home />);

    await waitFor(() => {
      expect(component.getAllByTestId("planet-card").length).toBe(10);
    });
    await waitFor(() => {
      expect(component.getByRole("button", { name: "Next" })).toBeEnabled();
    });

    act(() => {
      fireEvent.click(component.getByRole("button", { name: "Next" }));
    });

    await waitFor(() => {
      expect(component.getByText(planetsMock[10].name)).toBeInTheDocument();
    });
  });
});
