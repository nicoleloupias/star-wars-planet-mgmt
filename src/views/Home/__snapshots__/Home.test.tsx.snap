// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Home/> > should render succesfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-dark"
  >
    <div>
      <section
        class="css-16hn594"
      >
        <div
          class="css-ybip2u"
        />
        <h2
          class="chakra-heading css-113y03g"
        >
          Explore the galaxy
        </h2>
        <div
          class="css-1etnvhi"
        />
        <p
          class="chakra-text css-1kyjpe6"
        >
          Scroll
          <svg
            class="chakra-icon css-1j5r502"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              fill="currentColor"
            />
          </svg>
        </p>
      </section>
      <div
        class="css-b93e9j"
      >
        <section
          class="css-1bb2txy"
        >
          <form
            class="css-1654nw7"
          >
            <div
              class="css-nu2kwa"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="chakra-input__group css-1y0e7gb"
                  data-group="true"
                >
                  <input
                    class="chakra-input css-1rrwde8"
                    data-testid="search"
                    id="search"
                    name="search"
                    placeholder="Search"
                    value=""
                  />
                  <div
                    class="chakra-input__right-element css-1lds0jh"
                  >
                    <div
                      class="css-1rs9p06"
                    >
                      <button
                        aria-label="Search"
                        class="chakra-button css-1ni11ii"
                        type="submit"
                      >
                        <svg
                          aria-hidden="true"
                          class="chakra-icon css-onkibi"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                            fill="currentColor"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="chakra-select__wrapper css-awd3x6"
            >
              <select
                class="chakra-select css-szixr"
                name="sortBy"
              >
                <option
                  value=""
                >
                  Order by
                </option>
                <option
                  value="name"
                >
                  Name
                </option>
                <option
                  value="diameter"
                >
                  Diameter
                </option>
                <option
                  value="climate"
                >
                  Climate
                </option>
                <option
                  value="terrain"
                >
                  Terrain
                </option>
                <option
                  value="population"
                >
                  Habitants
                </option>
              </select>
              <div
                class="chakra-select__icon-wrapper css-iohxn1"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-select__icon"
                  focusable="false"
                  role="presentation"
                  style="width: 1em; height: 1em; color: currentColor;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </form>
          <div
            class="css-16138za"
          >
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/1"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-hpphqc"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Tatooine
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  10465
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  arid
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  desert
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  200000
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/2"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-99jkys"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Alderaan
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  12500
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  temperate
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  grasslands, mountains
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  2000000000
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/3"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-wb6ec1"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Yavin IV
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  10200
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  temperate, tropical
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  jungle, rainforests
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  1000
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/4"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-kpj6g7"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Hoth
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  7200
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  frozen
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  tundra, ice caves, mountain ranges
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  unknown
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/5"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-h5yfgu"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Dagobah
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  8900
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  murky
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  swamp, jungles
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  unknown
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/6"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-x46186"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Bespin
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  118000
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  temperate
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  gas giant
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  6000000
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/7"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-1k7y21q"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Endor
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  4900
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  temperate
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  forests, mountains, lakes
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  30000000
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/8"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-1a189rz"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Naboo
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  12120
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  temperate
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  grassy hills, swamps, forests, mountains
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  4500000000
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/9"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-17c27uo"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Coruscant
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  12240
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  temperate
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  cityscape, mountains
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  1000000000000
                </p>
              </div>
            </a>
            <a
              class="chakra-card css-ska63g"
              data-testid="planet-card"
              href="/10"
            >
              <div
                class="css-upfzb1"
              >
                <div
                  class="planet css-1uif6ec"
                />
              </div>
              <div
                class="cardDetails css-1weapz6"
              >
                <h3
                  class="chakra-heading css-1hna34j"
                >
                  Kamino
                </h3>
                <p
                  class="chakra-text css-0"
                >
                  Diameter: 
                  19720
                   
                  <span
                    class="css-1nvf5u9"
                  >
                    km
                  </span>
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Climate: 
                  temperate
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Terrain: 
                  ocean
                </p>
                <p
                  class="chakra-text css-0"
                >
                  Habitants: 
                  1000000000
                </p>
              </div>
            </a>
          </div>
          <button
            class="chakra-button css-1bvqs64"
            disabled=""
            type="button"
          >
            Prev
          </button>
          <button
            class="chakra-button css-1af772r"
            type="button"
          >
            Next
          </button>
        </section>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <section
      class="css-16hn594"
    >
      <div
        class="css-ybip2u"
      />
      <h2
        class="chakra-heading css-113y03g"
      >
        Explore the galaxy
      </h2>
      <div
        class="css-1etnvhi"
      />
      <p
        class="chakra-text css-1kyjpe6"
      >
        Scroll
        <svg
          class="chakra-icon css-1j5r502"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            fill="currentColor"
          />
        </svg>
      </p>
    </section>
    <div
      class="css-b93e9j"
    >
      <section
        class="css-1bb2txy"
      >
        <form
          class="css-1654nw7"
        >
          <div
            class="css-nu2kwa"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <div
                class="chakra-input__group css-1y0e7gb"
                data-group="true"
              >
                <input
                  class="chakra-input css-1rrwde8"
                  data-testid="search"
                  id="search"
                  name="search"
                  placeholder="Search"
                  value=""
                />
                <div
                  class="chakra-input__right-element css-1lds0jh"
                >
                  <div
                    class="css-1rs9p06"
                  >
                    <button
                      aria-label="Search"
                      class="chakra-button css-1ni11ii"
                      type="submit"
                    >
                      <svg
                        aria-hidden="true"
                        class="chakra-icon css-onkibi"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="chakra-select__wrapper css-awd3x6"
          >
            <select
              class="chakra-select css-szixr"
              name="sortBy"
            >
              <option
                value=""
              >
                Order by
              </option>
              <option
                value="name"
              >
                Name
              </option>
              <option
                value="diameter"
              >
                Diameter
              </option>
              <option
                value="climate"
              >
                Climate
              </option>
              <option
                value="terrain"
              >
                Terrain
              </option>
              <option
                value="population"
              >
                Habitants
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-iohxn1"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </form>
        <div
          class="css-16138za"
        >
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/1"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-hpphqc"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Tatooine
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                10465
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                arid
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                desert
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                200000
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/2"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-99jkys"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Alderaan
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                12500
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                temperate
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                grasslands, mountains
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                2000000000
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/3"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-wb6ec1"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Yavin IV
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                10200
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                temperate, tropical
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                jungle, rainforests
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                1000
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/4"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-kpj6g7"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Hoth
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                7200
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                frozen
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                tundra, ice caves, mountain ranges
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                unknown
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/5"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-h5yfgu"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Dagobah
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                8900
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                murky
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                swamp, jungles
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                unknown
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/6"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-x46186"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Bespin
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                118000
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                temperate
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                gas giant
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                6000000
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/7"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-1k7y21q"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Endor
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                4900
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                temperate
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                forests, mountains, lakes
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                30000000
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/8"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-1a189rz"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Naboo
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                12120
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                temperate
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                grassy hills, swamps, forests, mountains
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                4500000000
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/9"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-17c27uo"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Coruscant
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                12240
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                temperate
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                cityscape, mountains
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                1000000000000
              </p>
            </div>
          </a>
          <a
            class="chakra-card css-ska63g"
            data-testid="planet-card"
            href="/10"
          >
            <div
              class="css-upfzb1"
            >
              <div
                class="planet css-1uif6ec"
              />
            </div>
            <div
              class="cardDetails css-1weapz6"
            >
              <h3
                class="chakra-heading css-1hna34j"
              >
                Kamino
              </h3>
              <p
                class="chakra-text css-0"
              >
                Diameter: 
                19720
                 
                <span
                  class="css-1nvf5u9"
                >
                  km
                </span>
              </p>
              <p
                class="chakra-text css-0"
              >
                Climate: 
                temperate
              </p>
              <p
                class="chakra-text css-0"
              >
                Terrain: 
                ocean
              </p>
              <p
                class="chakra-text css-0"
              >
                Habitants: 
                1000000000
              </p>
            </div>
          </a>
        </div>
        <button
          class="chakra-button css-1bvqs64"
          disabled=""
          type="button"
        >
          Prev
        </button>
        <button
          class="chakra-button css-1af772r"
          type="button"
        >
          Next
        </button>
      </section>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
